cmd/kubeadm/app/cmd/config.go

```diff
@@ -519,7 +519,12 @@ func AddImagesCommonConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiv1beta1.In
 		&cfg.ClusterConfiguration.KubernetesVersion, "kubernetes-version", cfg.ClusterConfiguration.KubernetesVersion,
 		`Choose a specific Kubernetes version for the control plane.`,
 	)
-	flagSet.StringVar(featureGatesString, "feature-gates", *featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
-		"Options are:\n"+strings.Join(features.KnownFeatures(&features.InitFeatureGates), "\n"))
+	if knownFeatures := features.KnownFeatures(&features.InitFeatureGates); len(knownFeatures) > 0 {
+		flagSet.StringVar(featureGatesString, "feature-gates", *featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
+			"Options are:\n"+strings.Join(knownFeatures, "\n"))
+	} else {
+		flagSet.StringVar(featureGatesString, options.FeatureGatesString, *featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
+			"Options are:\n"+"No gates are available in this release.")
+	}
```

Comment:
i think this `if ...else` block should be made into an utility function here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/options/generic.go

also for the case where there are no FGs:
it should read:
```
flagSet.StringVar(featureGatesString, options.FeatureGatesString, *featureGatesString, "A set of key=value pairs that describe feature gates for various features. " +
	"No feature gates are available in this release.")
```

Reply:
@neolit123 Thank you very much! I've reorganized the code.